nodeJS/express/introduction_to_express.md

```diff
@@ -0,0 +1,98 @@
+
+### A request's journey
+
+Now that our server is up and running on port 3000, let's send it a request! In a browser, navigate to `http://localhost:3000/` (don't worry if you forget the slash `/` at the end; the browser will silently add it for you if so). This action tells the browser to send a `GET` request to the `/` route of whatever server is listening at port 3000 on our localhost (which is our Express server!) and display in the window whatever it receives in response.
+
+Whenever you navigate to any web URL this way, this is essentially what you are doing. Navigating to `https://theodinproject.com/paths` via the address bar is just telling the browser to send a `GET` request to the `/paths` route at `https://theodinproject.com`, then display what it receives in response.
+
+Once you navigate to `http://localhost:3000/`, you should see `Hello, world!` appear in the window. Magic, right?
+
+When our server receives our `GET` request, Express stores the request in a [request object](https://expressjs.com/en/4x/api.html#req). This request gets passed through a chain of functions we call `middleware functions` until eventually, a middleware function tells Express to respond to the request.
+
+In our example, the request comes through as a `GET` request to the `/` route. This matches the route handler we have in our `app.js` file.
+
+```javascript
+app.get("/", (req, res) => res.send("Hello, world!"));
+```
+
+We will discuss routes and route handlers in more detail in a later lesson, but to summarise the above line, it tells Express "if a `GET` request comes through to the `/` route, run the request through the following chain of middleware functions". Here, we only have a single function.
+
+If we had defined multiple route handlers, Express would pass the request through the first one to match the requested route. Order matters!
+
+Express takes the callback function we gave it and passes the request object into the first parameter (commonly named `req` by convention), and a [response object](https://expressjs.com/en/4x/api.html#res) into the second parameter (commonly named `res`). Our callback tells the response object to respond to the request by `.send`ing the string `"Hello, world!"`.
+
+There is no more code to run and the function returns. Since Express has been told to respond to the request, it ends the request here. Meanwhile, the browser receives our server's response and displays it on screen, which is our `"Hello, world!"` string. We could send nearly anything in response. We could even [tell Express to send a file](https://expressjs.com/en/api.html#res.sendFile).
```

Comment:
```suggestion
Since there is no more code to run, the function returns and Express responds to the request, effectively ending the request-response cycle. Meanwhile, the browser receives our server's response, which is our "Hello, world!" string, and displays it on the screen. We could send nearly anything in a response. We could even [tell Express to send a file](https://expressjs.com/en/api.html#res.sendFile).
```